internal/pkg/scorecard/helpers.go

```diff
@@ -35,3 +40,85 @@ func ResultsCumulative(results []TestResult) (earned, max int) {
 	}
 	return earned, max
 }
+
+// CalculateResult returns a ScorecardSuiteResult with the state and Tests fields set based on a slice of ScorecardTestResults
+func CalculateResult(tests []*scapiv1alpha1.ScorecardTestResult) *scapiv1alpha1.ScorecardSuiteResult {
+	scorecardSuiteResult := scapiv1alpha1.ScorecardSuiteResult{}
+	scorecardSuiteResult.Tests = tests
+	for _, test := range scorecardSuiteResult.Tests {
+		scorecardSuiteResult.TotalTests++
+		switch test.State {
+		case scapiv1alpha1.ErrorState:
+			scorecardSuiteResult.Error++
+		case scapiv1alpha1.PassState:
+			scorecardSuiteResult.Pass++
+		case scapiv1alpha1.PartialPassState:
+			scorecardSuiteResult.PartialPass++
+		case scapiv1alpha1.FailState:
+			scorecardSuiteResult.Fail++
+		}
+	}
+	return &scorecardSuiteResult
+}
+
+// TestSuitesToScorecardOutput takes an array of test suites and generates a v1alpha1 ScorecardOutput object with the
+// provided name, description, and log
+func TestSuitesToScorecardOutput(suites []*TestSuite, log string) *scapiv1alpha1.ScorecardOutput {
+	test := &scapiv1alpha1.ScorecardOutput{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ScorecardOutput",
+			APIVersion: "osdk.openshift.io/v1alpha1",
+		},
+		Log: log,
+	}
+	scorecardSuiteResults := []scapiv1alpha1.ScorecardSuiteResult{}
+	for _, suite := range suites {
+		results := []*scapiv1alpha1.ScorecardTestResult{}
+		for _, testResult := range suite.TestResults {
+			results = append(results, TestResultToScorecardTestResult(*testResult))
+		}
+		scorecardSuiteResult := CalculateResult(results)
+		scorecardSuiteResult.TotalScore = suite.TotalScore()
+		scorecardSuiteResult.Name = suite.GetName()
+		scorecardSuiteResult.Description = suite.GetDescription()
+		scorecardSuiteResult.Log = suite.Log
+		scorecardSuiteResults = append(scorecardSuiteResults, *scorecardSuiteResult)
+	}
+	test.Results = scorecardSuiteResults
+	return test
+}
+
+// TestResultToScorecardTestResult is a helper function for converting from the TestResult type to the ScorecardTestResult type
+func TestResultToScorecardTestResult(tr TestResult) *scapiv1alpha1.ScorecardTestResult {
+	sctr := scapiv1alpha1.ScorecardTestResult{}
+	sctr.State = tr.State
+	sctr.Name = tr.Test.GetName()
+	sctr.Description = tr.Test.GetDescription()
+	sctr.EarnedPoints = tr.EarnedPoints
+	sctr.MaximumPoints = tr.MaximumPoints
+	sctr.Suggestions = tr.Suggestions
+	if sctr.Suggestions == nil {
+		sctr.Suggestions = []string{}
+	}
+	stringErrors := []string{}
+	for _, err := range tr.Errors {
+		stringErrors = append(stringErrors, err.Error())
+	}
+	sctr.Errors = stringErrors
+	return &sctr
+}
+
+// UpdateState updates the state of a TestResult.
+func (res *TestResult) UpdateState() {
```

Comment:
I think this should be close to the TestResult struct IMO. having it here would be confusing and hard to find IMO.